src/Stryker.Core/Stryker.Core/Initialisation/InputFileResolver.cs

```diff
@@ -161,18 +166,31 @@ private IEnumerable<string> ExtractProjectFolders(string projectFilePath, bool f
             {
                 foreach (var sharedProject in new ProjectFileReader().FindSharedProjects(xDocument))
                 {
+                    var sharedProjectName = ReplaceMsbuildProperties(sharedProject, projectAnalyzerResult.Properties ?? EmptyProperties);
 
-                    if (!_fileSystem.File.Exists(_fileSystem.Path.Combine(projectDirectory, sharedProject)))
+                    if (!_fileSystem.File.Exists(_fileSystem.Path.Combine(projectDirectory, sharedProjectName)))
                     {
-                        throw new FileNotFoundException($"Missing shared project {sharedProject}");
+                        throw new FileNotFoundException($"Missing shared project {sharedProjectName}");
                     }
 
-                    var directoryName = _fileSystem.Path.GetDirectoryName(sharedProject);
+                    var directoryName = _fileSystem.Path.GetDirectoryName(sharedProjectName);
                     folders.Add(_fileSystem.Path.Combine(projectDirectory, directoryName));
                 }
             }
 
             return folders;
         }
+
+        private static string ReplaceMsbuildProperties(string value, IReadOnlyDictionary<string, string> properties)
+        {
+            try
+            {
+                return PropertyRegex.Replace(value, m => properties[m.Groups[1].Value]);
+            }
+            catch (KeyNotFoundException e)
+            {
+                throw new FileNotFoundException($"Cannot replace all properties in {value}", e);
```

Comment:
I think this should rather be a stryker input exception, as something is wrong in the input provided to stryker to run the test

Reply:
Will do, I wasn't sure which exception to use, in the end I didn't want the change the API surface so I ended up using whichever was thrown before the change...

Reply:
That is understandable, I don't believe we have properly documented the exception strategy.

Reply:
Updated to `StrykerInputException`, I took the liberty and added `innerException` to `StrykerInputException` ctor, if you don't want to keep track of the inner exception I'll revert it back.